src/main/java/org/osiam/ResourceServer.java

```diff
@@ -0,0 +1,220 @@
+        hikariConfig.setDriverClassName(driverClassName);
+        hikariConfig.setJdbcUrl(databaseUrl);
+        hikariConfig.setUsername(databaseUserName);
+        hikariConfig.setPassword(databasePassword);
+        return hikariConfig;
+    }
+
+    @Bean
+    @DependsOn("flyway")
+    public EntityManagerFactory entityManagerFactory() throws SQLException {
+
+        HibernateJpaVendorAdapter vendorAdapter = new HibernateJpaVendorAdapter();
+        vendorAdapter.setGenerateDdl(false);
+        vendorAdapter.setShowSql(false);
+        vendorAdapter.setDatabasePlatform(databasePlatform);
+
+        Properties jpaProperties = new Properties();
+        jpaProperties.put("hibernate.hbm2ddl.auto", "validate");
+        jpaProperties.put("hibernate.ejb.naming_strategy", "org.hibernate.cfg.ImprovedNamingStrategy");
+
+        LocalContainerEntityManagerFactoryBean factory = new LocalContainerEntityManagerFactoryBean();
+        factory.setJpaVendorAdapter(vendorAdapter);
+        factory.setPackagesToScan("org.osiam.storage.entities");
+        factory.setDataSource(dataSource());
+        factory.setJpaProperties(jpaProperties);
+        factory.afterPropertiesSet();
+
+        return factory.getNativeEntityManagerFactory();
+    }
+
+    @Bean
+    public JpaTransactionManager transactionManager() throws SQLException {
+        JpaTransactionManager txManager = new JpaTransactionManager();
+        txManager.setEntityManagerFactory(entityManagerFactory());
+        return txManager;
+    }
+
+    @Bean
+    public MetricRegistry metricRegistry() {
+        return new MetricRegistry();
+    }
+
+    @Bean
+    public HibernateExceptionTranslator hibernateExceptionTranslator() {
+        return new HibernateExceptionTranslator();
+    }
+
+    @Configuration
+    @EnableResourceServer
+    protected static class ResourceServerConfiguration extends ResourceServerConfigurerAdapter {
+
+        @Autowired
+        private AccessTokenValidationService accessTokenValidationService;
+
+        @Bean
+        public OAuth2AuthenticationEntryPoint createEntryPoint() {
+            OAuth2AuthenticationEntryPoint entryPoint = new OAuth2AuthenticationEntryPoint();
+            entryPoint.setRealmName("oauth2-authorization-server");
+            return entryPoint;
+        }
+
+        @Bean
+        public ShaPasswordEncoder passwordEncoder() {
+            ShaPasswordEncoder passwordEncoder = new ShaPasswordEncoder(512);
+            passwordEncoder.setIterations(1000);
+            return passwordEncoder;
+        }
+
+        @Override
+        public void configure(ResourceServerSecurityConfigurer resources) {
+            resources
+                    .resourceId("oauth2res")
+                    .tokenServices(accessTokenValidationService)
+                    .expressionHandler(new DefaultWebSecurityExpressionHandler());
+        }
+
+        @Bean
+        public OAuth2AccessDeniedHandler createOauthAccessDeniedHandler() {
```

Comment:
Shouldn't that be `oauthAccessDeniedHandler` to be consistent with the other `@Bean` declarations?